node/core/approval-voting/src/lib.rs

```diff
@@ -160,6 +160,7 @@ struct MetricsInner {
 	time_db_transaction: prometheus::Histogram,
 	time_recover_and_approve: prometheus::Histogram,
 	candidate_signatures_requests_total: prometheus::Counter<prometheus::U64>,
+	unapproved_candidates_in_unfinalized_chain: prometheus::Counter<prometheus::U64>,
```

Comment:
I think this should be a GaugeVec. Keep in mind that approval checking might take between 1-2 relay chain blocks so this would just rise up each call via `finality_target_with_longest_chain` by GRANDPA making the metric useless for differentiating between normal operation and failure.

I suggest to have 2 labels for it:
-  `count` to capture situations when approval voting continuously fails to approve certain number of candidates resulting in this going up
-  `depth`  to capture when just 1 single candidate was not approved while all rest are approved leading to a continuous increase of `depth`

Reply:
for depth, there already is `polkadot_parachain_approval_checking_finality_lag`